yt/visualization/plot_window.py

```diff
@@ -973,7 +979,8 @@ def setup_callbacks(self):
             CallbackMaker = callback_registry[key]
             callback = invalidate_plot(apply_callback(CallbackMaker))
             callback.__doc__ = CallbackMaker.__doc__
-            self.__dict__['annotate_'+cbname] = types.MethodType(callback,self)
```

Comment:
This is a reasonable refactor although it might be worth bringing this up on yt-dev.